src/receive_imf.rs

```diff
@@ -2385,12 +2379,19 @@ async fn has_verified_encryption(
     mimeparser: &MimeMessage,
     from_id: ContactId,
     to_ids: &[ContactId],
-    chat_type: Chattype,
 ) -> Result<VerifiedEncryption> {
     use VerifiedEncryption::*;
 
-    if from_id == ContactId::SELF && chat_type == Chattype::Single {
-        // For outgoing emails in the 1:1 chat, we have an exception that
+    // We do not need to check if we are verified with ourself.
+    let to_ids = to_ids
+        .iter()
+        .copied()
+        .filter(|id| *id != ContactId::SELF)
+        .collect::<Vec<ContactId>>();
+
+    if from_id == ContactId::SELF && to_ids.len() <= 1 {
+        // For outgoing emails in the 1:1 chat and groups with 2 members,
+        // we have an exception that
         // they are allowed to be unencrypted:
         // 1. They can't be an attack (they are outgoing, not incoming)
```

Comment:
Afaiu, verified chats are the way to protect against compromised servers as well, but if we allow this, we only protect from compromised servers of our contacts, but not from our own one. Finding the server a user can really trust isn't an easy challenge usually.

Reply:
Ok, but my concern is still here. Is there any document describing which kinds of attacks do verified chats protect from?

Reply:
Https://securejoin.delta.chat

Reply:
> What would the compromised own server do? Can trick the user into thinking that they sent some message when they didn't? Might be the case if the account is shared, but otherwise I don't see how this could work.

Just want to say that this is already fixed anyway :) Yes, it's common to save some things in the Saved messages like email addresses or URLs, so the server really could have a chance to trick the user.